### DISCUSSION TITLE
MQTT over a XB3 cellular module

### DISCUSSION THREAD

**Discussion by vpimark** [cesanta/mongoose#2886] [Ask questions, get answers] [answered]
I would like to use MQTT over serial to a XB3 cellular module.
I have configured the cellular module with at commands (ATDL91.121.93.94 and ATDE75B for ipaddress and port) and have connected to mqtt://test.mosquitto.org:1883
at that point I call mq_tcpip_init  with tcpInterface.driver setup with a tx and up functions as in the RNDIS example.  But do not ever achieve a MQTT connection.

What would be the correct way to do to this?

**Comment by scaprile** [comment, marked as answer]
"MQTT over serial" does not exist
I don't know what is "the RNDIS example" for you
I don't have the slightest idea of what an XB3 module is, nor of its commands.
You don't describe what you are trying to achieve at the system level, and what I presume is that you probably are trying to send IP inside a TCP/IP connection.
You don't describe your hardware, and I don't have the time to imagine all possible scenarios.
If you want to connect to a serial port, Mongoose is not the software you need, at least not today.

**Reply by vpimark** [reply]
I am trying to use Mongoose to establish an MQTT connection over a XB3-C-GM2-UT-001 cellular module from Digi International. I create a connection using AT commands, to mqtt://test.mosquitto.org:1883.  I can currently use Mongoose to do this over the ethernet phy on a stm32H.  I was wondering what to call to skip over the connecting with the URL,  since already have a connection through the AT commands, and still send all tcp/ip data to establish a MQTT connect.

**Reply by vpimark** [reply]
The RNDIS example is from your website. https://mongoose.ws/documentation/tutorials/rp2040/pico-rndis-dashboard/

**Reply by scaprile** [reply]
Mongoose is not the software you need, at least not today.
